### PR TITLE
conky: add lua-cairo binding

### PR DESCRIPTION
**Pull request by antonio-malcolm**
Title says it all (well, most of it)- this sets up the Lua-Cairo binding for Conky.

### CODE REVIEW

**Comment by antonio-malcolm** [comment]
Just to make mention of the change to lock Conky down to Lua v5.1, it's more stable when using just that version. I *have* had it working with v5.2, by adding tolua as a dependency to both toluapp and Conky, but if it works with v5.1, I see no need to require both libs.

Simple explanation is, toluapp was build as an extension of tolua, and will use tolua, if it's available. However, while the former is an extension of the latter, the latter was updated to support Lua v5.2, the former is stuck on v5.1, and it does not appear that either has seen activity in quite a few years.

There's been some discussion around updating Conky, to supply another, more current means of establishing the Lua-Cairo binding...

As for the added sed statements- upon the next Conky release, we may not need them (the maintainer merged two pull requests: one from me, another from an Arch contributor), which should ensure our package names are supported in ConkyPlatformChecks.cmake. For the time being, this should have us covered.

**Comment by antonio-malcolm** [comment]
Is there anything I need to change, with this?

**Comment by Vaelatern** [comment]
Reviewed this. Can you change the commit message to be `conky: add lua-cairo binding` so I can merge it?

File: srcpkgs/conky/template [hunk 1]
```diff
@@ -1,14 +1,16 @@
...
  -DCMAKE_BUILD_TYPE=Release -DMAINTAINER_MODE=ON -DDOC_PATH=share/${pkgname}
  -DBUILD_X11=ON -DBUILD_CURL=ON -DBUILD_XDBE=ON -DBUILD_RSS=ON -DBUILD_WEATHER_METAR=ON
- -DBUILD_WEATHER_XOAP=ON -DBUILD_IMLIB2=ON -DBUILD_WLAN=ON"
-hostmakedepends="cmake libtool pkg-config docbook2x man-db git"
-makedepends="lua-devel alsa-lib-devel libglib-devel libxml2-devel libXft-devel
+ -DBUILD_WEATHER_XOAP=ON -DBUILD_IMLIB2=ON -DBUILD_WLAN=ON -DBUILD_LUA_CAIRO=ON
+ -DBUILD_LUA_IMLIB2=ON -DBUILD_LUA_RSVG=ON -DBUILD_XSHAPE=ON"
+hostmakedepends="cmake libtool pkg-config docbook2x man-db git toluapp"
+makedepends="lua51-devel toluapp-devel cairo-devel glibc-devel librsvg-devel alsa-lib-devel
+ libglib-devel libxml2-devel libX11-devel libXext-devel libXfixes-devel libXft-devel
```
Comment (Vaelatern) [inline srcpkgs/conky/template]: Will be fine as `lua51-devel toluapp-devel glibc-devel librsvg-devel alsa-lib-devel libXft-devel`

**Comment by antonio-malcolm** [comment]
Sure thing- Title changed to "conky: add lua-cairo binding". Thanks!

**Comment by Vaelatern** [comment]
Please amend the message of the commit. Also pending review of which makedepends are needed.

File: srcpkgs/conky/template (hunk 1 above)
Comment (Vaelatern) [inline srcpkgs/conky/template]: Nevermind, not quite that simple.

**Comment by antonio-malcolm** [comment]
It looks like libXft-devel supplies everything (from looking through the dependency chain), except for libXfixes-devel and cairo-devel. Admittedly, I may have missed libXfixes being in the chain, somewhere, but, ifnothing else, shouldn't we include cairo-devel?

Thanks.

**Comment by Vaelatern** [comment]
```
makedepends="lua51-devel toluapp-devel glibc-devel librsvg-devel alsa-lib-devel
 libxml2-devel libXft-devel libXdamage-devel imlib2-devel wireless_tools-devel
 libcurl-devel ncurses-devel"
```

**Comment by antonio-malcolm** [comment]
Looks like libXdamage will provide libXfixes, and imlib2-devel will provide libxml2, but don't we still need Cairo? ConkyPlatformChecks.cmake checks for it, during the build, with option -DBUILD_LUA_CAIRO=ON. As soon as we can confirm this, I can update the template.

**Comment by Vaelatern** [comment]
`librsvg-devel` has `depends="gdk-pixbuf-devel cairo-devel libcroco-devel ${sourcepkg}>=${version}_${revision}"` so pulls in `cairo-devel`

**Comment by antonio-malcolm** [comment]
AH, ok. Missed that- I had looked at the rsvg template, but hadn't thought to look at the -devel block.
Ok, about a minute, to update...

Thanks!

File: srcpkgs/conky/template [hunk 2]
```diff
@@ -20,6 +22,11 @@ checksum=b4e66a71e8211d02b8dbe8cd4873af209af25326e63afd90c90ba9416c79cf21
 
 post_extract() {
 	sed -i '1iinclude(CheckIncludeFile)' cmake/ConkyPlatformChecks.cmake
+	sed -i 's/(LUA REQUIRED lua5.2 lua-5.2 lua>=5.1 lua5.1 lua-5.1)/(LUA REQUIRED lua5.1 lua-5.1)/g' cmake/ConkyPlatformChecks.cmake
+	sed -i 's/(APP_TOLUA NAMES tolua++ tolua++5.1 tolua++-5.1)/(APP_TOLUA NAMES toluapp)/g' cmake/ConkyPlatformChecks.cmake
+	sed -i 's/(FATAL_ERROR "Unable to find program '\''tolua++'\''")/(FATAL_ERROR "Unable to find program '\''toluapp'\''")/g' cmake/ConkyPlatformChecks.cmake
+	sed -i 's/(TOLUA_LIBS NAMES tolua++ tolua++5.1 tolua++-5.1)/(TOLUA_LIBS NAMES toluapp)/g' cmake/ConkyPlatformChecks.cmake
+	sed -i 's/(FATAL_ERROR "Unable to find tolua++ library")/(FATAL_ERROR "Unable to find toluapp library")/g' cmake/ConkyPlatformChecks.cmake
```
Comment (Vaelatern) [inline srcpkgs/conky/template]:
```
sed -i -e 's/(LUA REQUIRED lua5.2 lua-5.2 lua>=5.1 lua5.1 lua-5.1)/(LUA REQUIRED lua5.1 lua-5.1)/g' \
	 -e 's/(APP_TOLUA NAMES tolua++ tolua++5.1 tolua++-5.1)/(APP_TOLUA NAMES toluapp)/g' \
	 -e 's/(FATAL_ERROR "Unable to find program '\''tolua++'\''")/(FATAL_ERROR "Unable to find program '\''toluapp'\''")/g' \
	 -e 's/(FATAL_ERROR "Unable to find tolua++ library")/(FATAL_ERROR "Unable to find toluapp library")/g' \
	 -e 's/(TOLUA_LIBS NAMES tolua++ tolua++5.1 tolua++-5.1)/(TOLUA_LIBS NAMES toluapp)/g' cmake/ConkyPlatformChecks.cmake
```
to have just one sed call do all the changes to the same file.

**Comment by Vaelatern** [comment]
Good on the fixed commit message, by the way.

**Comment by antonio-malcolm** [comment]
Yup, figured it makes the most sense, for commit messages to match the pull request.

File: srcpkgs/conky/template (hunk 2 above)
Comment (Vaelatern) [inline srcpkgs/conky/template]:
```
post_extract() {
	sed -i \
		-e '1iinclude(CheckIncludeFile)' \
		-e 's/(LUA REQUIRED lua5.2 lua-5.2 lua>=5.1 lua5.1 lua-5.1)/(LUA REQUIRED lua5.1 lua-5.1)/g' \
		-e 's/(APP_TOLUA NAMES tolua++ tolua++5.1 tolua++-5.1)/(APP_TOLUA NAMES toluapp)/g' \
		-e 's/(FATAL_ERROR "Unable to find program '\''tolua++'\''")/(FATAL_ERROR "Unable to find program '\''toluapp'\''")/g' \
		-e 's/(TOLUA_LIBS NAMES tolua++ tolua++5.1 tolua++-5.1)/(TOLUA_LIBS NAMES toluapp)/g' \
		-e 's/(FATAL_ERROR "Unable to find tolua++ library")/(FATAL_ERROR "Unable to find toluapp library")/g' cmake/ConkyPlatformChecks.cmake
}
```

Comment (Vaelatern) [inline srcpkgs/conky/template]: Yeah, the above `post_extract()` function and I'll consider this ready to merge.

**Comment by antonio-malcolm** [comment]
Concatenated the sed commands.

**Comment by antonio-malcolm** [comment]
With the sed commands concatenated into one, this is failing on the ARM7 build test.

**Comment by Vaelatern** [comment]
Reran the arm test. Worked that time.

**Comment by antonio-malcolm** [comment]
Yup.

**Pull request merged**